File: Anul II/Medii de proiectare si programare/Laborator/lab07/Ubb.BikeContest.Server/ConcurrentServer.cs
using System.Net.Sockets;
using Ubb.BikeContest.Networking;

namespace Ubb.BikeContest.Services
{
    public abstract class ConcurrentServer : IServer
    {
        public ConcurrentServer(string host, int port) : base(host, port)
        { }

        public override void ProcessRequest(TcpClient client)
        {
            Thread t = CreateWorker(client);
            t.Start();
        }

        protected abstract Thread CreateWorker(TcpClient client);
    }
}

File: Anul II/Medii de proiectare si programare/Laborator/lab08/Java/server/src/main/java/com/ubb/server/ConcurrentServer.java
package com.ubb.server;

import java.net.Socket;

public abstract class ConcurrentServer extends Server {
    public ConcurrentServer(int port) {
        super(port);
    }

    protected void processRequest(Socket client) {
        Thread worker = createWorker(client);
        worker.start();
    }

    protected abstract Thread createWorker(Socket client);
}
